packages/react-dom/src/test-utils/ReactTestUtils.js

```diff
@@ -396,21 +393,17 @@ const ReactTestUtils = {
   SimulateNative: {},
 
   act(callback: () => Thenable) {
-    if (actContainerElement === null) {
-      if (__DEV__) {
-        // warn if we're trying to use this in something like node (without jsdom)
-        if (didWarnAboutActInNodejs === false) {
-          didWarnAboutActInNodejs = true;
-          warningWithoutStack(
-            typeof document !== 'undefined' && document !== null,
-            'It looks like you called ReactTestUtils.act(...) in a non-browser environment. ' +
-              "If you're using TestRenderer for your tests, you should call " +
-              'ReactTestRenderer.act(...) instead of ReactTestUtils.act(...).',
-          );
-        }
+    if (__DEV__) {
+      // warn if we're trying to use this in something like node (without jsdom)
+      if (didWarnAboutActInNodejs === false) {
+        didWarnAboutActInNodejs = true;
+        warningWithoutStack(
+          typeof document !== 'undefined' && document !== null,
```

Comment:
I know it already exists but structure is confusing because the variable is called `didWarnAboutActInNodejs` but it would be true if we did _not_ warn too.

I'd do this instead:

```js
if (typeof document === 'undefined' || document === null) {
  warningWithoutStack(didWarnAboutActInNodejs, '...');
  didWarnAboutActInNodejs = true;
}
```

Or even two levels of nesting and just `warning(false, ...)`

Reply:
yeah I'll clean this up

Reply:
Fixed by nuking the whole thing ha. updated PR description.